db.c

```diff
@@ -696,48 +696,42 @@ static void mongo_db_list_collections_command(zval *this_ptr, zval *options, int
 
 	php_mongo_command_cursor_init_from_document(db->link, cmd_cursor, connection->hash, cursor_env TSRMLS_CC);
 
-	for (
-		php_mongocommandcursor_load_current_element(cmd_cursor TSRMLS_CC);
-		php_mongocommandcursor_is_valid(cmd_cursor);
-		php_mongocommandcursor_advance(cmd_cursor TSRMLS_CC)
+	cmd_cursor->started_iterating = 1;
+
+	while (
+		php_mongocommandcursor_advance(cmd_cursor TSRMLS_CC) == SUCCESS &&
+		php_mongocommandcursor_is_valid(cmd_cursor)
 	) {
-		zval **collection_doc;
+		zval **collection_name, **collection_doc = &cmd_cursor->current;
```

Comment:
Please split the &cmd_cursor->current initialisation out of the declarations. This looked like a nice WTF :-)